src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/CodeActionEndpointTest.cs

```diff
@@ -799,7 +799,7 @@ private static IClientConnection CreateLanguageServer()
 
     private static RazorCodeDocument CreateCodeDocument(string text)
     {
-        var codeDocument = TestRazorCodeDocument.CreateEmpty();
+        var codeDocument = TestRazorCodeDocument.Create(text);
```

Comment:
If we don't pass text here and create empty RazorCodeDocument, it ends up containing empty Source (RazorSourceDocument), and any attempt to resolve positions fails. The unit tests were passing only because it's a single-line document and we never checked "character" (column) position in our code. After I added character check, almost all Code Action tests failed.